projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -350,6 +356,134 @@ describe('DatagridComponent', () => {
+                    };
+                    this.finder.detectChanges();
+                });
+
+                describe('top position', () => {
+                    it('shows no buttons with nothing selected', function(this: HasFinderAndGrid): void {
+                        expect(this.clrGridWidget.getButtons().length).toEqual(0);
+                    });
+
+                    it('shows two buttons with record 1 is selected', function(this: HasFinderAndGrid): void {
+                        this.clrGridWidget.selectRow(0);
+                        expect(this.clrGridWidget.getButtons()).toEqual(['Add', 'Other', '']);
+                    });
+
+                    it('shows one button when record 2 is selected', function(this: HasFinderAndGrid): void {
+                        this.clrGridWidget.selectRow(1);
+                        expect(this.clrGridWidget.getButtons()).toEqual(['Add', '']);
+                    });
+
+                    it('responds when the user presses a contextual button', function(this: HasFinderAndGrid): void {
+                        const spy = jasmine.createSpy('clickHanlder');
+                        this.finder.hostComponent.buttonConfig.contextualButtons.buttons[0].handler = spy;
+                        this.clrGridWidget.selectRow(0);
+                        this.clrGridWidget.pressContextualButton(0);
+                        expect(spy).toHaveBeenCalledWith([mockData[0]]);
+                    });
+                });
+
+                describe('row position', () => {
+                    beforeEach(function(this: HasFinderAndGrid): void {
+                        this.finder.hostComponent.buttonConfig.contextualButtons.position =
+                            ContextualButtonPosition.ROW;
+                        this.finder.detectChanges();
+                    });
+
+                    it('shows two buttons next to record 1', function(this: HasFinderAndGrid): void {
+                        expect(this.clrGridWidget.getRowButtons(0)).toEqual(['Add', 'Other', '']);
+                    });
+
+                    it('shows one button next to record 2', function(this: HasFinderAndGrid): void {
+                        expect(this.clrGridWidget.getRowButtons(1)).toEqual(['Add', '']);
+                    });
+
+                    it('responds when the user presses a contextual button', function(this: HasFinderAndGrid): void {
+                        const spy = jasmine.createSpy('clickHanlder');
+                        this.finder.hostComponent.buttonConfig.contextualButtons.buttons[0].handler = spy;
+                        this.finder.detectChanges();
+                        this.clrGridWidget.pressContextualButtonAtRow(0, 0);
+                        expect(spy).toHaveBeenCalledWith([mockData[0]]);
+                    });
+
+                    it('calculates the correct number of buttons to show on a row', function(this: HasFinderAndGrid): void {
```

Comment:
Please avoid correct. What does it mean to be correct? The text of the it should try to match the text of the expect.  

it adds CSS classes to set the correct width based on the number of buttons